Emby.Naming/Video/VideoListResolver.cs

```diff
@@ -141,7 +141,13 @@ private static List<VideoInfo> GetVideosGroupedByVersion(List<VideoInfo> videos,
                 {
                     if (group.Key)
                     {
-                        videos.InsertRange(0, group.OrderByDescending(x => x.Files[0].FileNameWithoutExtension.ToString(), new AlphanumericComparator()));
+                        videos.InsertRange(0, group
+                            .OrderBy(x =>
```

Comment:
Don't all items of the group have the same resolution?
I think we just need to traverse groups in descending order and use accending order in the group.

Reply:
No, that misunderstanding is exactly why https://github.com/jellyfin/jellyfin/pull/12620 was broken and wrong. The actual way how it works is that, all items with a resolution tag are in the same group

Reply:
Ah, `IsMatch` for grouping (with and without resolution).
Can we use `Match` there?

Reply:
> Can we use Match there?

If we do that then probably we don't need a grouping at all, I can try

> Also, shouldn't we do OrderByDescending for the resolution and then ascending for the name?

It should not matter? Currently it is ascending by name (after the resolution tag) then descending by resolution

> Also, we are comparing resolutions as strings.

We are not. The unit test will fail if we are. The part is compared as strings are the names after the resolution tag and the resolution tag is already compared with AlphanumericComparator

Reply:
Well I tested that not using grouping at all and it does not seems to break test locally, let's see if CI is happy with that.

Reply:
> If we do that then probably we don't need a grouping at all, I can try

👍

> It should not matter? Currently it is ascending by name (after the resolution tag) then descending by resolution

But this is `OrderBy` for the resolution (ascending, resolution as is) and `ThenByDescending` for the rest (when the resolution doesn't matter - it is the same for the compared values). 👀

> The unit test will fail if we are.

I haven't found a test for version order with multiple resolutions. 🤷‍♂️

> the resolution tag is already compared with AlphanumericComparator

The resolution order is controlled by the first `OrderBy`.
`ThenBy...` is applied to the versions with the same resolution after `OrdeBy`.
`AlphanumericComparator` is only applied to `ThenBy...` when the resolution doesn't matter.
No?

Reply:
> I haven't found a test for version order with multiple resolutions. 🤷‍♂️

In `MultiVersionTests.cs`. That's the test that the previous PR was failing

> The resolution order is controlled by the first OrderBy.
ThenBy... is applied to the versions with the same resolution after OrdeBy.
AlphanumericComparator is only applied to ThenBy... when the resolution doesn't matter.
No?

No. The first `OrderBy` is used to sort the part after resolution, the later `ThenBy` is used to sort the resolution. I know the regex code is hard to understand but this is how it currently works. The resolution regex is used to match the resolution string and the substring index is used to only take the part after the resolution tag. For example `2160p Remux` will have `2160p` be sorted with `AlphanumericComparator` in the `ThenBy` and ` Remux` will be sorted with the first OrderBy

Reply:
> In `MultiVersionTests.cs`. That's the test that the previous PR was failing

Oh. I was searching in the `VideoListResolver` test.

But we probably need another test:
```
- B
- A
- C
... - 1080p B
... - 1080p A
... - 2160p B
... - 720p A
```
ordered to this (according to the current ordering version)
```
... - 1080p A
... - 720p A
... - A
... - 2160p B
... - 1080p B
... - B
... - C
```

Do we support these (resolution after version name)?
```
... - B 1080p
... - A 2160p
```

> and the substring index is used to only take the part after the resolution tag.

I missed that part and was thinking about sorting the resolution regardless of its position in the version name.

Reply:
Well I don't think current one is correct and perhaps the grouping cannot be omitted for more complicated cases

Reply:
Alright I reverted the part that omitted the grouping. Basically, the ones without resolution should appear behind the one with resolution, and resolution should leading the version name, so that the versions with the same resolutions are combined together in the list. See `TestMultiVersion13` for the expected behavior.

Reply:
> Do we support these (resolution after version name)?

I don't think so? But if we want to we need to modify the logic a little bit.